### DISCUSSION TITLE
Request cap

### DISCUSSION THREAD

**Discussion by SpyderAlba** [tweepy/tweepy#1445] [Questions] [answered]
I am working on a project using Tweepy and the premium API. However, I keep hitting the request cap long before hitting the maximum tweets. Is this a product of how tweepy connects to the API?

![Screen Shot 2020-10-02 at 1 47 14 PM](https://user-images.githubusercontent.com/3988262/94964156-8d882180-04b6-11eb-8678-2a78638046ee.png)

```python
search_words='@superdrug OR @SuperdrugHelp OR @SDrug_Northwich OR @LloydsPharmacy OR @echo_pharmacy OR @lookfantastic OR @LookfantasticCS OR @beautybay OR @beautybayhelp OR @asda OR @AsdaServiceTeam'
date_since='202008010000'
#date_to='202010021936'
#### premium search
tweets=tp.Cursor(api.search_full_archive,environment_name='dev',\
                 query=search_words, fromDate=date_since,\
                 toDate=date_to).items(6000)
cnt = 0
with open('comp_tweets.json','a') as file:
    for tweet in tweets:
        cnt += 1
        json.dump(tweet._json, file)
        file.write('\n')
        print(tweet.created_at)
        date_to =tweet.created_at.strftime('%Y%m%d%H%M')
file.close()
print(f"Added {cnt} more tweets")
```

**Comment by Harmon758** [comment, marked as answer]
You can see [in the Developer Portal](https://developer.twitter.com/en/account/subscriptions/search-fullarchive), that the Premium Search Tweets: Full Archive package has a set 1.25M monthly Tweet cap, regardless of which level of usage. 

Note though, that the [premium search API](https://developer.twitter.com/en/docs/twitter-api/premium/search-api/api-reference/premium-search) that [`API.search_full_archive`](https://tweepy.readthedocs.io/en/latest/api.html#API.search_full_archive) uses has a `maxResults` parameter:
> The maximum number of search results to be returned by a request. A number between 10 and the system limit (currently 500, 100 for Sandbox environments). By default, a request response will return 100 results.

Since you're not setting this parameter, it defaults to 100, and you're receiving 100 results per request, rather than the maximum possible 500 results per request if you were to set it to 500.

The 1.25M monthly Tweet cap matches the maximum number of Tweets that can be retrieved with the highest level of usage that you can select: 2,500 requests per month * 500 results per request = 1,250,000 results per month.
